data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java

```diff
@@ -137,31 +137,21 @@ private Future<Void> reconcileEgress(final DataPlaneContract.Contract contract)
       futures.add(
         this.egressReconcilerListener.onUpdateEgress(newResource, newEgress)
           // If we fail to update the egress we can't put it in the cache.
-          .onFailure(r -> egresses.remove(uid))
           .onSuccess(r -> this.cachedEgresses.put(newEgress.getUid(), new SimpleImmutableEntry<>(newEgress, newResource)))
           .onFailure(cause -> logFailure("Failed to reconcile [onUpdateEgress] egress ", newEgress, cause, generation))
       );
     });
 
     // We want to complete the future, once all futures are complete, so use join.
-    return CompositeFuture.join(futures)
-      .onComplete(r -> {
```

Comment:
so we do not need the inlined clean up, on completion, due to the above changes on the "cache"?